transport/src/main/java/io/netty/channel/socket/nio/NioServerSocketChannel.java

```diff
@@ -198,5 +220,37 @@ private NioServerSocketChannelConfig(NioServerSocketChannel channel, ServerSocke
         protected void autoReadCleared() {
             clearReadPending();
         }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public <T> T getOption(ChannelOption<T> option) {
+            if (option == SO_MAXCONNECTIONS) {
```

Comment:
yeah, i consider only nioserversocketchannel when i wrote the code. let me rewrite it to make it more universal